torch/utils/hooks.py

```diff
@@ -61,6 +74,8 @@ def __setstate__(self, state) -> None:
             if len(state) < 3
             else weakref.ref(OrderedDict() if state[2] is None else state[2])
         )
+        # TODO can we actually restore module_ref after unpickling? Do we care?
```

Comment:
Please do not merge PRs with random TODOs in utils or nn. This code is exercised in many ways compared to dynamo code and this kind of things really matter.

Reply:
Yes see comments below. There are clear BC issues. I think there is some mis-categorization of this as an "optimization" (implying that it's fine if we're not doing a perfect job) even though it is actually critical for correctness (and so we have to do a perfect job here).